kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -67,7 +67,8 @@ public class TransactionImpl
     private final Path dataPath;
     private final Path logPath;
     private final Protocol protocol;
-    private final Metadata metadata;
+    private Metadata metadata;
+    private boolean shouldUpdateMetadata;
```

Comment:
can we revert these changes? Looks like these aren't getting updated anywhere.

Reply:
It will be updated for the following PRs. But it's fine to keep it `final` here.